File: ProyectoMysql/server/EntityLayer/OrdenCompraDetalleEntity.py
from datetime import datetime
from pydantic import BaseModel
from Utilidades.Enumerado.ProcessActionEnum import ProcessActionEnum

class OrdenCompraDetalleSaveModel(BaseModel):
    OrdenCompraDetalleId: int 
    OrdenCompraId: int 
    MercaderiaId: int 
    UnidadMedidaId: int 
    CantidadSolicitado: float 
    CantidadComprado: float 
    CantidadFaltante : float 
    PrecioUnitario: float 
    Action: ProcessActionEnum

class OrdenCompraDetalleItemModel:
    OrdenCompraDetalleId: int 
    OrdenCompraId: int 
    MercaderiaId: int 
    UnidadMedidaId: int 
    CantidadSolicitado: float 
    CantidadComprado: float 
    CantidadFaltante : float 
    PrecioUnitario: float 
    NomProducto: str 
    CategoriaId: int 
    CodigoUM: str 
    Stock: float 
    def Cargar(_DB):
        c =  OrdenCompraDetalleItemModel()
        c.OrdenCompraDetalleId = _DB["OrdenCompraDetalleId"] 
        c.OrdenCompraId = _DB["OrdenCompraId"] 
        c.MercaderiaId = _DB["MercaderiaId"] 
        c.UnidadMedidaId = _DB["UnidadMedidaId"] 
        c.CantidadSolicitado = _DB["CantidadSolicitado"] 
        c.CantidadComprado = _DB["CantidadComprado"] 
        c.CantidadFaltante = _DB["CantidadFaltante"] 
        c.PrecioUnitario = _DB["PrecioUnitario"] 
        c.NomProducto = _DB["NomProducto"] 
        c.CategoriaId = _DB["CategoriaId"] 
        c.CodigoUM = _DB["CodigoUM"] 
        c.Stock = _DB["Stock"] 
        
        return c

File: ProyectoMysql/Api/LogisticStorage/LogisticStorage.EntityLayer/OrdenCompraDetalleEntity.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace LogisticStorage.EntityLayer
{
    public partial class OrdenCompraDetalleEntity : BaseEntityObject
    {
        public OrdenCompraDetalleEntity()
        {
            this.OrdenCompraDetalleId = 0;
            this.OrdenCompraId = 0;
            this.MercaderiaId = 0;
            this.UnidadMedidaId = 0;
            this.CantidadSolicitado = 0;
            this.CantidadComprado = 0;
            this.CantidadFaltante = 0;
            this.PrecioUnitario = 0;
            this.NomProducto = String.Empty;
            this.CategoriaId = 0;
            this.CodigoUM = String.Empty;
            this.Stock = 0;
        }
        public Int32 OrdenCompraDetalleId { get; set; }
        public Int32 OrdenCompraId { get; set; }
        public Int32 MercaderiaId { get; set; }
        public Int32 UnidadMedidaId { get; set; }
        public Decimal CantidadSolicitado { get; set; }
        public Decimal CantidadComprado { get; set; }
        public Decimal CantidadFaltante { get; set; }
        public Decimal PrecioUnitario { get; set; }

    }
}
